src/ui/DynamicHierarchicalFacet/DynamicHierarchicalFacet.ts

```diff
@@ -364,10 +375,25 @@ export class DynamicHierarchicalFacet extends Component implements IDynamicHiera
 
   private updateAppearance() {
     this.header.toggleCollapse(this.isCollapsed);
+    this.toggleSearchDisplay();
     $$(this.element).toggleClass('coveo-dynamic-hierarchical-facet-collapsed', this.isCollapsed);
     $$(this.element).toggleClass('coveo-hidden', !this.values.allFacetValues.length);
   }
 
+  private toggleSearchDisplay() {
```

Comment:
This method is copied from [DynamicFacets'](https://github.com/coveo/search-ui/blob/master/src/ui/DynamicFacet/DynamicFacet.ts#L770)